mda_openbabel_converter/OpenBabelParser.py

```diff
@@ -4,26 +4,119 @@
-        can be converted to a MDAnalysis Topology?)
+        (i.e., is it a valid OpenBabel OBMol that can be converted to a 
+        MDAnalysis Topology?)
         """
-        pass
+        try:
+            import openbabel as OB
+        except ImportError:
+            return False
+        else:
+            return isinstance(thing, OB.OBMol)
 
-    def parse(self, **kwargs):
+    def parse(self, filename: OBMol, **kwargs):
         """
         Accepts an OpenBabel OBMol and returns a MDAnalysis Topology. Will need
         to extract the number of atoms, number of residues, number of segments,
-        atom_residue index, residue_segment index and other attributes from the
-        OBMol to initialise a new Topology. 
+        atom_residue index, residue_segment index and all of the atom's 
+        relevant attributes from the OBMol to initialise a new Topology. 
         """
+        format = 'OPENBABEL'
+
+        self.atoms = []
+        self.n_atoms = 0
+        self.residues = []
+        self.n_residues = 0
+        self.segments = []
+        self.n_segments = 0
+
+        obmol = filename
+
+        # Atoms
+        names = []
+        chiralities = []
+        resnums = []
+        resnames = []
+        elements = []
+        masses = []
+        charges = []
+        aromatics = []
+        ids = []
+        atomtypes = []
+        segids = []
+        altlocs = []
+        chainids = []
+        icodes = []
+        occupancies = []
+        tempfactors = [] # B factor; not supported by OB
+
+        if obmol.GetFirstAtom().equals(None):
+            return Topology(n_atoms = 0,
+                            n_res = 0,
+                            n_seg = 0,
+                            attrs=None,
+                            atom_resindex=None,
+                            residue_segindex=None)
+
+        for atom in OB.OBMolAtomIter(obmol):
+            # need to add handling incase attributes are invalid or null in OBMol
+            # names.append(atom.GetType()) #char -> nothing for name in OBMol? Is name required to make MDA Atom?
+            atomtypes.append(atom.GetType()) #char
+            ids.append(atom.GetIdx()) #int
+            masses.append(atom.GetExactMass()) #double -> what about atom.GetAtomicMass()??; which is better?
+            if not atom.GetExactMass().equals(atom.GetAtomicMass()):
+                warnings.warn(
+                    f"Exact mass and atomic mass of atom (ID: {atom.GetIdx})
+                      not equal. Be aware of isotopes, which are NOT supported 
+                      by MDAnalysis.")
+            charges.append(atom.GetPartialCharge()) #int (or use atom.GetFormalCharge()?)
+
+            # convert atomic number to element
+            elements.append(OBElementTable.GetSymbol(atom.GetAtomicNumber())) #char
+
+            resid = atom.GetResidue() # null if no residue
+            if resid.equals(None):
+                warnings.warn(
```

Comment:
This is kind of a meta point as well, but if its `None` its `None` so to speak. 

You are only responsible for translating the two representations, if they need residues in their MDAnalysis representation from an OBMol, they need to put them in at the Obabel level or with MDAnalysis later.

Reply:
Ok makes sense. So it would be best to just set to null and warn the user that they may need to set it later? And leave it at that?

Reply:
Yes exactly. Assuming you mean `None` by null.

Reply:
Ok got it, and yep